Integrations/SecBI/SecBI.py

```diff
@@ -0,0 +1,203 @@
+        :return:  A dictionary representation of the SecBI Incident found to contain the supplied host
+        """
+        params = get_dict_without_none(host=host, fields=SECBI_INCIDENT_FIELDS)
+        raw_incident = self._http_request(method='GET', url_suffix=SECBI_API_ENDPOINT_INCIDENT, params=params)
+        return as_secbi_incident(raw_incident)
+
+    def test_module(self):
+        """
+        Performs basic get request to test SecBI system
+        """
+        return self._http_request(method='GET', url_suffix=SECBI_API_ENDPOINT_STATUS)
+
+
+def test_module_command(client: SecBIClient):
+    """
+    This is the call made when pressing the integration test button.
+    Performs basic get request to test SecBI system
+    :param client: The SecBI client to use for the sanity test
+    """
+    results = client.test_module()
+    if 'status' in results and results['status'] == 'ACTIVE':
+        demisto.results('ok')
+    raise DemistoException("Failed connection test of SecBI. Please check your SECBI_API_URL and SECBI_API_KEY")
+
+
+def secbi_get_incidents_list_command(client: SecBIClient, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    SecBI Get All Incident IDs command
+    :param client: The SecBI client to use
+    :param args: The Demisto args
+    :return: Content for return_outputs()
+    """
+    query = args.get('query', None)
+    limit = args.get('limit', 100)
+    incidents_list = client.secbi_get_incidents_list(query, limit)
+
+    formatted = [dict(id=s) for s in incidents_list]
+    human_readable = tableToMarkdown('List of SecBI Incidents', formatted)
+    entry_context = {
+        'SecBI.IncidentsList': incidents_list
+    }
+    return human_readable, entry_context, incidents_list
+
+
+def secbi_get_incident_command(client: SecBIClient, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    SecBI Get Incident command
+    :param client: The SecBI client to use
+    :param args: The Demisto args
+    :return: Content for return_outputs()
+    """
+    incident_id = args.get('incident_id', None)
+    incident_data = client.secbi_get_incident(incident_id)
+
+    human_readable = tableToMarkdown('SecBI incident ID "{}"'.format(incident_id), incident_data)
+    entry_context = {
+        'SecBI.Incident(val.id == obj.id)': incident_data
+    }
+    return human_readable, entry_context, incident_data
+
+
+def secbi_get_incident_by_host_command(client: SecBIClient, args: Dict) -> Tuple[str, Dict, Dict]:
+    """
+    SecBI Get Incident by Host command
+    :param client: The SecBI client to use
+    :param args: The Demisto args
+    :return: Content for return_outputs()
+    """
+    host = args.get('host', None)
+    incident_data = client.secbi_get_incident_by_host(host)
+
+    human_readable = tableToMarkdown('SecBI incident by host "{}"'.format(host), incident_data)
+    entry_context = {
+        # ID comparison might be problematic if the host changes Incidents and the ID changes
```

Comment:
Not sure I understand that comment, can you please elaborate?

Reply:
This has to do with the way hosts are handled within SecBI. A given `host` may move change incidents between any two given SecBI "runs" (or digestions of data), thus the ID (and content) will change although they have the same desired host.
This might cause some problems if a user is relying on the same ID for the same host.

Reply:
In that case, we will have to change the unique field from `ID` to one that does not change

Reply:
This is by design according to the use-case defined. The comment was more for us to take note in case the use-case definition ever changes.